src/vivarium/framework/results/context.py

```diff
@@ -353,55 +359,81 @@ def get_observations(self, event: Event) -> list[Observation]:
         """
         return [
             observation
-            for observations in self.observations[event.name].values()
+            for observations in self.grouped_observations[event.name].values()
             for observation in observations
             if observation.to_observe(event)
         ]
 
+    def get_stratifications(self, event: Event) -> list[Stratification]:
+        """Get all stratifications for a given event.
+
+        Parameters
+        ----------
+        event
+            The current Event.
+
+        Returns
+        -------
+            A list of Stratifications for the given event. Only includes stratifications
+            whose `to_observe` method returns True.
+        """
+        return [
+            self.stratifications[stratification_name]
+            for _, stratification_names in self.grouped_observations[event.name]
+            if stratification_names is not None
+            for stratification_name in stratification_names
```

Comment:
We want to exclude stratifications here if all observations in using it have their `to_observe()` return `False`.

This will also duplicate Stratifications if they appear multiple times here.